daemon/config.go

```diff
@@ -87,3 +95,84 @@ func getDefaultNetworkMtu() int {
 	}
 	return defaultNetworkMtu
 }
+
+// Parse the remapped root (user namespace) option, which can be one of:
+//   username            - valid username from /etc/passwd
+//   username:groupname  - valid username; valid groupname from /etc/group
+//   uid                 - 32-bit unsigned int valid Linux UID value
+//   uid:gid             - uid value; 32-bit unsigned int Linux GID value
+//
+//  If no groupname is specified, and a username is specified, an attempt
+//  will be made to lookup a gid for that username as a groupname
+//
+//  If names are used, they are mapped to the appropriate 32-bit unsigned int
+func parseRemappedRoot(usergrp string) (int, int, error) {
```

Comment:
I know it was there because I remember looking at that first.. let me dig into my February/March self and remind myself why I would waste time re-writing it :-/